odoo/orm/fields_relational.py

```diff
@@ -569,6 +621,59 @@ def _check_sudo_commands(self, comodel):
             return comodel.sudo(False).with_user(comodel.env.transaction.default_env.uid)
         return comodel
 
+    def condition_to_sql(self, model: BaseModel, alias: str, field_expr: str, operator: str, value, query: Query) -> SQL:
+        assert field_expr == self.name, "Supporting condition only to field"
+        model._check_field_access(self, 'read')
+        comodel = model.env[self.comodel_name]
+        if self.auto_join:
+            # bypass access rules for auto_join
+            comodel = comodel.sudo()
+
+        # update the operator to 'any'
+        if operator in ('in', 'not in'):
+            operator = 'any' if operator == 'in' else 'not any'
+        assert operator in ('any', 'not any'), \
+            f"Relational field {self} expects 'any' operator"
+        exists = operator == 'any'
+
+        # check the value and execute the query
+        if isinstance(value, OrderedSet) and False in value:
+            # If there are nulls to be checked, the condition is inversed.
+            #  in (False, 1) => not any (id not in (1))
+            exists = not exists
+            comodel = comodel.sudo()
+            value = Domain('id', 'not in', value - {False})
+        elif isinstance(value, SQL):
+            # wrap SQL into a simple query
+            comodel = comodel.sudo()
+            value = Domain('id', 'any', value)
+        coquery = self._get_query_for_condition_value(model, comodel, value)
+        return self._condition_to_sql_relational(model, alias, exists, coquery, query)
+
+    def _get_query_for_condition_value(self, model: BaseModel, comodel: BaseModel, value) -> Query:
+        """ Return Query run on the comodel with the field.domain injected."""
+        field_domain = Domain(self.get_domain_list(model))
+        if isinstance(value, Domain):
+            domain = value & field_domain
+            query = comodel.with_context(**self.context)._search(domain)
+            if not isinstance(query, Query):
+                query = comodel.browse(query)._as_query(ordered=False)
+            return query
+        if isinstance(value, Query):
+            # add the field_domain to the query
+            domain = field_domain._optimize(comodel)
+            if not domain.is_true():
+                # TODO should clone/copy Query value
+                value.add_where(domain._to_sql(comodel, value.table, value))
+            return value
+        if isinstance(value, OrderedSet):
+            assert False not in value, "False value should be removed during preparation"
+            return comodel.browse(value).filtered_domain(field_domain)._as_query(ordered=False)
```

Comment:
Mhhh that's new.

Reply:
What exactly? It handles `('user_ids', 'in', [1, 2])`. We apply the field domain as in other cases.

Reply:
Before we didn't check the domain for this case but it seems fair to do it anyway.

Reply:
After checking, it seems this allows to skip checking the related table in a m2m relationship. I will check what can be simplified.

Reply:
I handled this case in the caller function so that it provides a `Query` instead.